nipype/workflows/rsfmri/fsl/tests/test_resting.py

```diff
@@ -89,16 +89,19 @@ def test_create_resting_preproc(self, mock_node, mock_realign_wf):
         # assert
         expected_file = os.path.abspath(self.out_filenames['components_file'])
         with open(expected_file, 'r') as components_file:
-            components_data = [line.split() for line in components_file]
-            num_got_components = len(components_data)
-            assert (num_got_components == self.num_noise_components
-                    or num_got_components == self.fake_data.shape[3])
-            first_two = [row[:2] for row in components_data[1:]]
-            assert first_two == [['-0.5172356654', '-0.6973053243'], [
-                '0.2574722644', '0.1645270737'
-            ], ['-0.0806469590',
-                '0.5156853779'], ['0.7187176051', '-0.3235820287'],
-                                 ['-0.3783072450', '0.3406749013']]
+            components_data = [line.split()
+                               for line in components_file.read().splitlines()]
+        num_got_components = len(components_data)
+        if not (num_got_components == self.num_noise_components or
+                num_got_components == self.fake_data.shape[3]):
+            raise AssertionError()
+        first_two = [row[:2] for row in components_data[1:]]
+        if first_two != [['-0.5172356654', '-0.6973053243'],
+                         ['0.2574722644', '0.1645270737'],
+                         ['-0.0806469590', '0.5156853779'],
+                         ['0.7187176051', '-0.3235820287'],
+                         ['-0.3783072450', '0.3406749013']]:
+            raise AssertionError()
```

Comment:
I think all of this can be reverted. What's the motivation here? It makes sense to be keep only the `components_data` line inside the `with` block, but all of the rest of these changes don't make much sense to me.

Reply:
The motivation was that [codacy complained about the use of `assert` -- see 'fixed issues' tab](https://app.codacy.com/app/nipype/nipype/pullRequest?prid=3061026). I'll go ahead and revert and see if it raises the same concern again.

Reply:
Yeah, codacy is not turning out to be very helpful, IMO. Whatever general policy one might have about assertions, they're completely reasonable to have in tests.